Framework/API/inc/MantidAPI/IFunction.h

```diff
@@ -648,6 +650,13 @@ class MANTID_API_DLL IFunction {
   /// Get status of parameter
   [[nodiscard]] virtual ParameterStatus getParameterStatus(size_t i) const = 0;
```

Comment:
I think its generally a good idea to use it on functions that have a high-computational cost, such as calculations or functions containing loops - because we don't want unnecessary calls which waste time and resource.

I also think that it should probably be a default in c++ that a return value is `nodiscard`, and instead there should be a flag called `[[sometimes-discard]]`, but obviously this isn't possible for backwards compatibility reasons. I don't know what the convention is in Mantid, but if it were ever possible to have a perfect codebase I would say that every function that returns a value would have `[[nodiscard]]` (apart from the functions that should allow discard)